packages/core/src/amazonqGumby/telemetry/codeTransformTelemetry.ts

```diff
@@ -90,3 +105,79 @@ export const javapOutputToTelemetryValue = (javapCommandLineOutput: string) => {
 }
 
 export const calculateTotalLatency = (startTime: number): number => Date.now() - startTime
+
+type CodeTransformAPIresponse =
+    | CreateUploadUrlResponse
+    | ResumeTransformationResponse
+    | StopTransformationResponse
+    | GetTransformationPlanResponse
+    | StartTransformationResponse
+    | GetTransformationResponse
+
+export class TelemetryHelper {
+    private static get sessionId() {
+        return CodeTransformTelemetryState.instance.getSessionId()
+    }
+
+    static async callApi<T extends CodeTransformAPIresponse>(parameters: {
+        apiCall: () => Promise<PromiseResult<T, AWSError>>
+        apiName: CodeTransformApiNames
+        errorReason: string
+        uploadId?: string
+        jobId?: string
+        uploadFileByteSize?: number
+        setJobFailureMetadata: boolean
+        shouldAttachJobFailureMetadataOnError: boolean
+    }) {
+        const apiStartTime = Date.now()
+        let result = undefined
+        try {
+            result = await parameters.apiCall()
+            if (parameters.setJobFailureMetadata && result?.$response.requestId !== undefined) {
+                transformByQState.setJobFailureMetadata(` (request ID: ${result.$response.requestId})`)
+            }
+            return result
+        } catch (e: any) {
+            let errorMessage = (e as Error).message
+            getLogger().error(`CodeTransformation: ${parameters.apiName} error: = ${errorMessage}`)
+            if (parameters.shouldAttachJobFailureMetadataOnError) {
+                errorMessage += ` -- ${transformByQState.getJobFailureMetadata()}`
+            }
+            telemetry.codeTransform_logApiError.emit({
+                codeTransformApiNames: parameters.apiName,
+                codeTransformSessionId: this.sessionId,
+                codeTransformApiErrorMessage: errorMessage,
+                codeTransformRequestId: e?.requestId ?? telemetryUndefined,
+                codeTransformJobId: parameters.jobId ?? telemetryUndefined,
+                result: MetadataResult.Fail,
+                reason: parameters.errorReason,
+            })
+            throw e // pass along error to callee
+        } finally {
+            let uploadId = parameters.uploadId
+            try {
+                uploadId = (result as CreateUploadUrlResponse).uploadId
+            } catch (_) {
+                //noop
+            }
```

Comment:
try/catch is not necessary here. Just use `.?` operator.
that also allows use of `const`.

```suggestion
            const uploadId = (result as CreateUploadUrlResponse)?.uploadId ?? parameters.uploadId
```